core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -739,6 +739,9 @@ private[spark] object SparkConf extends Logging {
     (name.startsWith("spark.auth") && name != SecurityManager.SPARK_AUTH_SECRET_CONF) ||
     name.startsWith("spark.rpc") ||
     name.startsWith("spark.network") ||
+    // We need SSL configs to propagate as they may be needed for RPCs.
+    // Passwords are propagated separately though.
+    (name.startsWith("spark.ssl") && !name.contains("Password")) ||
```

Comment:
Dont you need the password in order to load the keys ?

Reply:
We do, and we pass them over environment variables. `isExecutorStartupConf` is used for arguments that are passed in the command line invocation to start the executor. If we passed the password there, anyone on the system using `ps` could see the password in plaintext, which we don't want. There is an existing test which fails, due to exactly this bug, that I had to adjust in this PR.

Reply:
Sounds good, thanks for clarifying